lib/puppet/provider/dependency.rb

```diff
@@ -1,5 +1,5 @@
 require 'puppet/provider'
-require 'puppet/provider/null_provider'
+require './null_provider'
```

Comment:
The `./` will cause this to load relative to the cwd if such a file exists, which is a security issue. I guess this should be `require_relative 'null_provider'`? I'm not sure how `require_relative` behaves exactly...